### PR TITLE
Fix nix build on OSX

### PR DESCRIPTION
**Pull request by jmatsushita**
cryptobox and zauth wouldn't build on OSX because the wrong path was included in the `dylib`.

The symptom was that the `cryptobox-c` haskell library build would fail with:
```
cryptobox-haskell                > running .stack-work/dist/x86_64-osx-nix/Cabal-3.0.1.0/build/System/CryptoBox_hsc_make failed (exit code -6)
cryptobox-haskell                > rsp file was: ".stack-work/dist/x86_64-osx-nix/Cabal-3.0.1.0/build/System/hsc2hscall42850-3.rsp"           
cryptobox-haskell                > output file:".stack-work/dist/x86_64-osx-nix/Cabal-3.0.1.0/build/System/CryptoBox.hs"                      
cryptobox-haskell                > command was: .stack-work/dist/x86_64-osx-nix/Cabal-3.0.1.0/build/System/CryptoBox_hsc_make  >.stack-work/dist/x86_64-osx-nix/Cabal-3.0.1.0/build/System/CryptoBox.hs
cryptobox-haskell                > error: dyld: Library not loaded: /private/tmp/nix-build-cryptobox-c-2019-06-17.drv-0/source/target/x86_64-apple-darwin/release/deps/libcryptobox.dylib
cryptobox-haskell                >   Referenced from: /private/tmp/stack-3f8b56a5eca0af97/cryptobox-haskell-0.1.1/.stack-work/dist/x86_64-osx-nix/Cabal-3.0.1.0/build/System/CryptoBox_hsc_make
cryptobox-haskell                >   Reason: image not found
```

```
$ otool -l /nix/store/hzj1xl80qq3xg7fxd6l3adx0ahil3pyb-cryptobox-c-2019-06-17/lib/libcryptobox.dylib
...
Load command 3
          cmd LC_ID_DYLIB
      cmdsize 144
         name /private/tmp/nix-build-cryptobox-c-2019-06-17.drv-0/source/target/x86_64-apple-darwin/release/deps/libcryptobox.dylib (offset 24)
   time stamp 1 Thu Jan  1 01:00:01 1970
...
```

The fix proposed in [this post](https://discourse.nixos.org/t/patchelf-for-mach-o-shared-libraries/562/6) fixed the problem.

```
$ otool -l /nix/store/w71z1pnxxx6wi69brl5pdzssdpa54yam-cryptobox-c-2019-06-17/lib/libcryptobox.dylib
...
Load command 3
          cmd LC_ID_DYLIB
      cmdsize 120
         name /nix/store/w71z1pnxxx6wi69brl5pdzssdpa54yam-cryptobox-c-2019-06-17/lib/libcryptobox.dylib (offset 24)
   time stamp 1 Thu Jan  1 01:00:01 1970
      current version 0.0.0
...
```

### CODE REVIEW

**Review by akshaymankar — APPROVED** [review]
Looks good to my noob eyes.

**Comment by jmatsushita** [comment]
Adding @lucendio as I heard you tried OSX Nix things before.

File: nix/overlays/wire-server.nix [hunk 1]
```diff
@@ -14,7 +14,13 @@ self: super: {
           sha256 = "1i9dlhw0xk1viglyhail9fb36v1awrypps8jmhrkz8k1bhx98ci3";
         };
         cargoSha256 = "0zs8ibv7rinrrzp9naxd7yak7kn1gp3pjb3g8i4wf7xw2hkkq81z";
+
+        patchLibs = super.lib.optionalString super.stdenv.isDarwin ''
+            install_name_tool -id $out/lib/libcryptobox.dylib $out/lib/libcryptobox.dylib
```
Comment (arianvp) [inline nix/overlays/wire-server.nix:19]: and this was absolutely needed? I thought it fixed itself by just bumping `nixpkgs` no?

The fact that `buildRustPackage` outputs binaries with broken paths sounds like something we should report as a bug upstream.

**Comment by lucendio** [comment]
> Adding @lucendio as I heard you tried OSX Nix things before.

due to my lack of experience with nix and Haskell on macos (and in general), I'm not equipped to review and give feedback.

File: nix/overlays/wire-server.nix (hunk 1 above)
Comment (jmatsushita) [reply nix/overlays/wire-server.nix:19]: That was PBUK. Paths looked ok.... but not when I scrolled up. And it didn't build until I made that change.

It's a known issue apparently https://discourse.nixos.org/t/patchelf-for-mach-o-shared-libraries/562/6 there's even a hook for it `fixDarwinDylibNames`.

**Comment by jmatsushita** [comment]
All good @lucendio thanks!

**Pull request merged**